src/NServiceBus.Gateway/Gateway.cs

```diff
@@ -139,9 +160,9 @@ static ConfigurationBasedSiteRouter GetConfigurationBasedSiteRouter(FeatureConfi
             return new ConfigurationBasedSiteRouter(sites);
         }
 
-        static void RegisterHttpListenerInstaller(FeatureConfigurationContext context, IManageReceiveChannels channelManager)
+        static void RegisterHttpListenerInstaller(FeatureConfigurationContext context, IManageReceiveChannels channelManager, bool enableHttpListener)
```

Comment:
@danielmarbach This flag allows us to disable the HTTP listener when custom channels are used. Previous implementation would not configure the `INeedToInstallSomething` implementation that started the listener when custom channels were used, which would lead to endpoint startup failure. 

I assume this would be done as part of #48 but I needed it earlier.

Does this make sense?